src/minichain/network.py

```
@@ -0,0 +1,1075 @@
+                break
+            blocks.append(block)
+
+        connection = self._connections.get(peer_id)
+        if connection is None:
+            return
+        response = self._sync_blocks_payload(start_height=from_height, blocks=blocks)
+        await self._write_message(connection.writer, response)
+
+    async def _handle_sync_blocks(self, peer_id: str, message: dict[str, object]) -> None:
+        if message.get("protocol") != SYNC_PROTOCOL_ID:
+            raise NetworkError("sync_blocks protocol id mismatch")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)